### PR TITLE
Completed calculator project

### PR DESCRIPTION
**Pull request by vikshab**
I have completed Breakfast, Lunch, Dinner and half of Dessert level. My calculator project has some sound affect, so make sure your volume is on:)

### CODE REVIEW

File: CalculatorG.rb [hunk 1]
```diff
@@ -0,0 +1,237 @@
+# Project Calculator (gold)
+
+puts "Choose from the list what operation you would like to do:\n "
+puts " add (+)\n sub (-)\n mult (*)\n div(/)\n exponent (^, **)\n mod (%)"
+user_input = gets.chomp
+
+# Function for addition……………………………………………………………………………………………………………………………………………………
```
Comment (kariabancroft) [inline CalculatorG.rb:7]: Good use of comments to separate it out. In a more complicated example you could use the comments to describe the overall functionality of the method.

**Comment by kariabancroft** [comment]
This looks great! So impressed that you used methods!

You could also DRY (don't repeat yourself) up the code by consolidating the logic that retrieves the first_number, second_number and result_type since that is the same in each method.

File: CalculatorG.rb [hunk 2]
```diff
@@ -0,0 +1,237 @@
...
+def add
+    puts "Ok, lets do addition. Type in first number"
+    first_number = gets.chomp
+        while first_number != "0" && first_number.to_f == 0
```
Comment (kariabancroft) [inline CalculatorG.rb:11]: Adding a comment here would be good to understand exactly what case these conditionals are checking for.

File: CalculatorG.rb [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+   sub
+elsif  user_input == "mult" || user_input == "*"
+   mult
+elsif user_input == "div" || user_input == "/"
+   div
+elsif user_input == "exponent" || user_input == "^" || user_input == "**"
+   exponent
+elsif user_input == "mod" || user_input == "%"
+   mod
+ else
```
Comment (kariabancroft) [inline CalculatorG.rb]: Watch for extra spaces in your indentation - I think there is an extra space here at the beginning of the line

**Comment by kariabancroft** [comment]
Great use of loops to ensure that the user provides a valid input for the first, second and result

**Comment by kariabancroft** [comment]
| Rating | Content           | Description
|--------|-------------------|-----------------------------------------------------------
|    2    | Variables                | Variables names are logical, clear, and in snake case. Variables are not needlessly assigned or reassigned.
|    1    | Loops                    | While loops control the flow of the code, keeping the user on track, and continuing the program until an appropriate end point.
|    1    | Data Types               | Hashes, arrays, strings, symbols, numbers, and booleans used appropriately.
|    2    | Conditionals & Operators | Logical and mathematical operators guide program flow efficiently and properly deal with edge cases (Gold).

File: CalculatorG.rb (hunk 3 above)
Comment (vikshab) [inline CalculatorG.rb]: Did you mean extra space on the 235 line? before else?

File: CalculatorG.rb (hunk 2 above)
Comment (vikshab) [inline CalculatorG.rb:11]: These conditionals are checking for correct user input , the input can be just numbers,  no letters are accepted

**Pull request merged**